### PR TITLE
Avoid Klaxxon dependency in Android Logins API (fixes #222)

### LINKED ISSUE #222: Remove Klaxon dependency in logins kotlin
This was requested by android-components. There's JSON parsing in the android stdlib, and if we want to use libs AFAICT they would rather us use gson or mochi (according to comments in https://github.com/mozilla-mobile/android-components/pull/676).

I think we'll try to use the stdlib stuff, to avoid issues coordinating dependencies. This shouldn't be hard, but I'm writing it down so it doesn't fall off my radar.

### PR DESCRIPTION
**Pull request by thomcc**

### CODE REVIEW

File: logins-api/android/library/src/main/java/org/mozilla/sync15/logins/ServerPassword.kt [hunk 1]
```diff
@@ -47,5 +50,53 @@ class ServerPassword (
...
+                    return null
+                }
+            }
+
+            return ServerPassword(
+                    id = jsonObject.getString("id"),
+
+                    hostname = jsonObject.getString("hostname"),
+                    password = jsonObject.getString("password"),
+                    username = stringOrNull("username"),
```
Comment (ncalexan) [inline logins-api/android/library/src/main/java/org/mozilla/sync15/logins/ServerPassword.kt]: This is what `optString` is for: https://developer.android.com/reference/org/json/JSONObject.html#optString(java.lang.String,%20java.lang.String)

File: logins-api/android/library/src/main/java/org/mozilla/sync15/logins/ServerPassword.kt [hunk 2]
```diff
@@ -47,5 +50,53 @@ class ServerPassword (
...
+
+                    timesUsed = jsonObject.getInt("timesUsed"),
+
+                    timeCreated = jsonObject.getLong("timeCreated"),
+                    timeLastUsed = jsonObject.getLong("timeLastUsed"),
+                    timePasswordChanged = jsonObject.getLong("timePasswordChanged")
+            )
+        }
+
+        fun fromJSON(jsonText: String): ServerPassword {
```
Comment (ncalexan) [inline logins-api/android/library/src/main/java/org/mozilla/sync15/logins/ServerPassword.kt]: I don't know a lot about Kotlin but it's weird to me that this doesn't include exception information.

File: logins-api/android/library/src/main/java/org/mozilla/sync15/logins/ServerPassword.kt (hunk 1 above)
Comment (thomcc) [reply logins-api/android/library/src/main/java/org/mozilla/sync15/logins/ServerPassword.kt]: For some reason I forgot that Java strings could be null (Kotlin ones can't, FWIW)

File: logins-api/android/library/src/main/java/org/mozilla/sync15/logins/ServerPassword.kt (hunk 2 above)
Comment (thomcc) [reply logins-api/android/library/src/main/java/org/mozilla/sync15/logins/ServerPassword.kt]: Yep, Kotlin doesn't check exceptions.

**Pull request merged**